app/src/lib/markdown-filters/mention-filter.ts

```diff
@@ -36,7 +36,7 @@ export class MentionFilter implements INodeFilter {
   private readonly dotsAtEndOfLine = /\.+$/
 
   // non-word character except dot or `
-  private readonly nonWordExceptDotOrBackTick = /[^0-9a-zA-Z_.`]/
+  private readonly nonWordExceptDotOrBackTickOrHyphen = /[^0-9a-zA-Z_.`-]/
```

Comment:
For the purposes of a username, the hyphen is a word character. This is intending to exclude things that could be in a username characters (signifying the end of username characters). I ran into it because I made an org called tidy-dev-org. Then when I referenced @tidy-dev-org/the-a-team, the @tidy-dev part would be caught and linked by our mention filter. It shouldn't because the regex is meant not to catch things that look like @something/. 
![image](https://user-images.githubusercontent.com/75402236/159012785-27d661b1-7b92-4783-9c10-e7535c0b035d.png)


Also to note, I originally put the call order of the TeamMentionFilter after the Mention Filter and simply swapping this prevented this problem because the team mention was caught first. Thus, I decided to put the TeamMentionFilter first in case there were any other regex fallacies.